### PR TITLE
Periodic report

### PR DESCRIPTION
**Pull request by dshulyak**

### CODE REVIEW

File: solar/solar/orchestration/graph.py [hunk 1]
```diff
@@ -157,3 +157,21 @@ def report_topo(uid):
...
+    """
+    start_time = time.time()
+
+    while start_time + timeout >= time.time():
+        dg = get_graph(uid)
+        not_finished = [n for n in dg if dg.node[n]['status']
+                        in (states.PENDING.name, states.INPROGRESS.name)]
+        yield len(not_finished), len(dg.nodes())
+        if not_finished == []: return
+        time.sleep(interval)
```
Comment (pigmej) [inline solar/solar/orchestration/graph.py]: I don't like that sleep there. It should be on consumer side (CLI in this case).

File: solar/solar/orchestration/graph.py [hunk 2]
```diff
@@ -157,3 +157,21 @@ def report_topo(uid):
...
+
+
+def wait_finish(uid, timeout, interval=3):
+    """Wait finish will periodically load graph and check if there is no
+    PENDING or INPROGRESS
+    """
+    start_time = time.time()
+
+    while start_time + timeout >= time.time():
+        dg = get_graph(uid)
```
Comment (pigmej) [inline solar/solar/orchestration/graph.py]: we should probably think how we could avoid getting whole graph always, but only changed things.

**Comment by loles** [comment]
+1

File: solar/solar/orchestration/graph.py [hunk 3]
```diff
@@ -157,3 +160,23 @@ def report_topo(uid):
...
+        summary = Counter()
+        summary.update({s.name: 0 for s in states})
+        summary.update([s['status'] for s in dg.node.values()])
+        yield summary
+        if summary[states.PENDING.name] + summary[states.INPROGRESS.name] == 0:
+            return
+        time.sleep(interval)
```
Comment (loles) [inline solar/solar/orchestration/graph.py]: can you move this sleep to cli?

Comment (dshulyak) [inline solar/solar/orchestration/graph.py]: i will need to repeat condition from 177-178 lines in cli

what the problem with this sleep?

Comment (pigmej) [inline solar/solar/orchestration/graph.py]: Sleep shouldn't be in library, it should be on consumer side (CLI in this situation).

**Pull request merged**